src/Storages/StorageLog.cpp

```diff
@@ -452,10 +455,20 @@ void LogSink::writeData(const NameAndTypePair & name_and_type, const IColumn & c
             const auto & data_file = *data_file_it->second;
             const auto & columns = metadata_snapshot->getColumns();
 
+            CompressionCodecPtr compression;
+            if (name_and_type.name == BlockNumberColumn.name)
+            {
+                auto data_bytes_size = BlockNumberColumn.type->getSizeOfValueInMemory();
+                compression = getCompressionCodecDelta(data_bytes_size);
```

Comment:
Compression codec delta without additional compression on top of it doesn't do anything. We need something like `Delta + ZSTD` or `Delta + LZ4`